### DISCUSSION TITLE
How should we handle heading levels?

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/guided-navigation#12] [General]
In ARIA, heading levels are communicated using `aria-level` on either `heading` or `treegrid`: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-level

How should we handle this information in Guided Navigation Documents?

I see two potential ways of handling this:

- duplicating the `heading` role to cover `<h1>` through `<h6>` and the equivalent usage of `aria-level`
- or the introduction of a new property specifically for this purpose

In our current implementations based on [the Content Iterator API](https://github.com/readium/architecture/blob/proposal/content-iterator/proposals/000-content-iterator.md#textstyle-enum), this is handled using an integer, which is similar to ARIA's approach.

**Comment by mickael-menu** [comment]
I'm in favor of having an extensibility point where we could have a dedicated property like the `aria-level` one. That would simplify checking for any heading, and we won't be stuck with 6 levels for all formats. (Although PDF is also using 6 levels AFAIK).
